### ISSUE TITLE
Module build failed: TypeError: files.forEach is not a function

### ISSUE THREAD

**Issue by IAMtheIAM** [epegzz/sass-vars-loader#18]
After updating 2.2.0 to 3.0.0:

```
Module build failed: TypeError: files.forEach is not a function
    at exports.default (C:\Source\node_modules\@epegzz\sass-vars-loader\dist\utils\watchFilesForChanges.js:6:9)
    at Object.exports.default (C:\node_modules\@epegzz\sass-vars-loader\dist\sassVarsLoader.js:13:38)
```

The file exists, but webpack cannot resolve it. Is there a breaking change?

**Comment by epegzz** [comment]
What node version are you running on? The latest sass-vars-loader requires node `v4.8.6`? Hope that's not an issue. I could lower the requirement with Babel but I thought Node v4.8.6 is a reasonable target nowadays.

**Comment by IAMtheIAM** [comment]
I am running 6.10.0. Npm 5.5.1

I will check my config again and update you with what i find,.

**Comment by epegzz** [comment]
Second guess would be, that you specified `files` in the config, but it's not an Array :)

**Comment by epegzz** [comment]
Closing this issue, feel free to re-open if the issue still exists.

**Comment by IAMtheIAM** [comment]
The issue just resurfaced.

```ts
const sassVarsConfig = querystring.stringify({
   vars : JSON.stringify({
      susyIsDevServer: susyIsDevServer,
      env            : METADATA.ENV
      // fullHost: METADATA.fullHost
   }), // Or use 'files" object to specify vars in an external .js or .json file
   files: [path.resolve(Helpers.paths.appRoot + '/assets/styles/sass-js-variables.js')]
});
```
That is my config. It is an array. What else might cause this?

```
Module build failed: TypeError: files.forEach is not a function
    at exports.default (C:\Source\frontend\node_modules\@epegzz\sass-vars-loader\dist\utils\watchFilesForChanges.js:6:9)
    at Object.exports.default (C:\Source\frontend\node_modules\@epegzz\sass-vars-loader\dist\sassVarsLoader.js:14:38)
 @ ./src/app-components/app/app.style.scss 4:14-526
 @ ./src/app-components/app/app.component.ts
 @ ./src/app-components/app/app.spec.ts
 @ ./src \.spec\.ts
 @ ./config/karma.entry.js
```

**Comment by epegzz** [comment]
Okay, don't use `JSON.stringify` :) The `env` parameter must be a JS object, not a string.

**Comment by epegzz** [comment]
If the issue should still persist even without the `JSON.stringify`, then try a `console.log(path.resolve(Helpers.paths.appRoot + '/assets/styles/sass-js-variables.js'))`, then paste the output of the log in a terminal with an `ls` command like this: `ls /the/result/of/console_log` and see if the file actually exisis

**Comment by IAMtheIAM** [comment]
I ran the command and the file does exist. I also removed the JSON.stringify and still files.foreach is not a function.

**Comment by IAMtheIAM** [comment]
I fixed it. I was using `querystring.stringify` and then in module.rules i used `@epegzz/sass-vars-loader?' + sassVarsConfig`. This worked in 2.x, but in 3.x it failed. I removed `querystring.stringify` (and JSON.stringify) and switched to 
```
               {
                  loader : '@epegzz/sass-vars-loader',
                  options: sassVarsConfig
               }
```
and it works.

**Comment by epegzz** [comment]
Hm right. I wonder if the lib should actually support query string parameters. 🤔

**Comment by IAMtheIAM** [comment]
It would be good as a fallback support, some people still use that. But load options is the new standard. Maybe it could just throw an error if query string is given, otherwise I don't think it is too hard to parse query string params right?
